core/src/mindustry/entities/comp/PayloadComp.java

```diff
@@ -162,4 +162,34 @@ void contentInfo(Table table, float itemSize, float width){
             table.image(p.icon(Cicon.small)).size(itemSize).padRight(pad);
         }
     }
+
+    @Override
+    public void destroy(){
+        //if this unit was carrying other units, spawn them but at reduced health
+        payloads.each(payload -> {
+            if(payload instanceof UnitPayload p){
+                Unit u = p.unit;
+
+                //drop all units, let ground units die on walls
+                Fx.unitDrop.at(this);
+
+                if(Vars.net.client()){
+                    //clear removed state of unit so it can be synced
+                    Vars.netClient.clearRemovedEntity(p.unit.id);
+                }else{
+                    //clients do not drop payloads
+                    u.set(this);
+                    u.trns(Tmp.v1.rnd(Mathf.random(2f)));
+                    u.rotation(rotation);
+                    //reset the ID to a new value to make sure it's synced
+                    u.id = EntityGroup.nextId();
+                    //decrement count to prevent double increment
+                    if(!u.isAdded()) u.team.data().updateCount(u.type, -1);
+                    //take damage from the landing
+                    u.health = u.maxHealth * u.type.crashDurability;
```

Comment:
doesn't take into account units that already had some damage, they damaged to or healed up to 10% health 🤔

Reply:
Then let's make it take damage instead.